### PR TITLE
Display V2

### PR DESCRIPTION
**Pull request by DalwynWasTaken**
Repurpose Display to use the more up-to-date rendering principles by using Text and the philosophies it has. 

Align enum has been moved to Text where it is actually used. Align has been added to Text to actually work alongside Display

Order enum has been renamed from UP and DOWN to NORMAL and REVERSED to indicate the changes made internally to rendering. Order now changes the order but not the position of the Display.

RegisterType enum has been removed as draw is now called by the module creator instead of internally. Removed a bunch of references to Display since it no longer internally renders everything.

DisplayLine has been removed as it was a glorified Text with some triggers.

Fixed a known memory sink when setting lines with a string. Adding and setting now should use a more optimized way of setting the text.

### CODE REVIEW

**Comment by DalwynWasTaken** [comment]
If you want me to write the stuff that would impact CT module developers into the migration docs, please tell me.

This is a drastic change to the way Display used to function but it allows for a lot more flexibility. 
DisplayLine has triggers attached to itself, this could be something interesting to add to Texts in general?

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/Display.kt [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        "}"
+
+    enum class Background {
+        NONE, FULL
+    }
+
+    enum class Order {
+        REVERSED, NORMAL
+    }
+}
```
Comment (DalwynWasTaken) [inline src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/Display.kt:188]: TODO: Add an extra line here

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Text.kt [hunk 2]
```diff
@@ -65,16 +64,17 @@ class Text {
...
-            is String -> Display.Align.valueOf(align.uppercase())
-            is Display.Align -> align
-            else -> Display.Align.LEFT
+            is String -> Align.valueOf(align.uppercase())
+            is Align -> align
+            else -> Align.LEFT
         }
     }
 
+
```
Comment (DalwynWasTaken) [inline src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Text.kt:77]: Remove this extra line

**Review by DalwynWasTaken — COMMENTED** [review]
Formatting fixup

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Text.kt [hunk 3]
```diff
@@ -126,6 +147,11 @@ class Text {
 
     @JvmOverloads
     fun draw(x: Float? = null, y: Float? = null) = apply {
+        draw(x, y, null, null)
+    }
+
+    @JvmOverloads
```
Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Text.kt:153]: Remove `@JvmOverloads` here

**Review by mattco98 — CHANGES_REQUESTED** [review]

**Pull request merged**